submissions/usides/js-dom/dist/js/main.js

```diff
@@ -0,0 +1,197 @@
+    charges:
+      '- Suspected in mouses assassination with a rifle <br>- mouse killer for money - hitcat',
+    portret: 'crim_06.jpg',
+    files: ['crim_06_e01.jpg', 'crim_06_e02.jpg'],
+  },
+];
+
+const sideNavMenu = document.getElementById('side-nav-menu');
+const sideMenu = document.getElementById('side-nav');
+const menuBtn = document.getElementById('menu-btn');
+const nextBtn = document.getElementById('next-btn');
+let current = 0;
+
+document.addEventListener('DOMContentLoaded', function () {
+  loadMenu();
+  printData(current);
+});
+
+menuBtn.addEventListener('click', function (e) {
+  e.preventDefault();
+  sideMenu.classList.toggle('hidden');
+  this.classList.toggle('active');
+});
+
+nextBtn.addEventListener('click', function (e) {
+  e.preventDefault();
+  current++;
+  printData(current);
+  makeActive(current);
+  if (current == suspects.length - 1) current = -1;
+});
+
+sideNavMenu.addEventListener('click', function (e) {
+  e.preventDefault();
+  if (e.target.nodeName !== 'A') return;
+  window.scrollTo({
+    top: 0,
+    left: 0,
+    behavior: 'smooth',
+  });
+  let arrNum = suspects.findIndex(
+    (item) => item.number === Number(e.target.id.slice(1)),
+  );
+  printData(arrNum);
+  makeActive(arrNum);
+  current = arrNum;
+  sideMenu.classList.add('hidden');
+  menuBtn.classList.remove('active');
+});
+
+function loadMenu() {
+  const fragment = document.createDocumentFragment();
+  for (let i = 0; i < suspects.length; i++) {
+    const newElem = document.createElement('li');
+    newElem.innerHTML = `<a href="#" class="menu-link" id="p${
+      suspects[i].number
+    }">${labelCreate.call(suspects[i])}</a>`;
+    if (i == 0) newElem.firstElementChild.classList.add('active');
+    fragment.appendChild(newElem);
+  }
+  sideNavMenu.appendChild(fragment);
+  current = 0;
+}
+
+function printData(perp) {
+  let holders = document.querySelectorAll('[data-holder]');
+  for (let i = 0; i < holders.length; i++) {
+    holders[i].innerHTML = suspects[perp][holders[i].dataset.holder];
+  }
+  document
+    .getElementById('crim-portret')
+    .setAttribute('src', `./img/${suspects[perp].portret}`);
+  document.getElementById('crim-label').textContent = labelCreate.call(
+    suspects[perp],
+  );
+  document.getElementById('evidence-set').innerHTML = '';
+  const fragment = document.createDocumentFragment();
+  for (let i = 0; i < suspects[perp].files.length; i++) {
+    const newElem = document.createElement('div');
+    newElem.innerHTML = `<img src="./img/${
+      suspects[perp].files[i]
+    }" alt="Case photo"><p>File ${i + 1}</p>`;
+    newElem.classList.add('evidence-item');
+    fragment.appendChild(newElem);
+  }
+  document.getElementById('evidence-set').appendChild(fragment);
+}
+
+function labelCreate() {
+  return `#${this.number}-${this.gender.slice(0, 1)}-${this.nick}`;
+}
+
+function makeActive(menuItem) {
```

Comment:
Your approach is totally ok here. But I'd recommend you simplifying this function to something like this:
``` javascript
function makeActive(target) {
    const activeMenuLink = document.querySelector('.menu-link.active');
    if (activeMenuLink) {
        activeMenuLink.classList.remove('active');
    }
    target.classList.add('active');
}
```

Reply:
Done